### ISSUE TITLE
Constructor signature incorrect when inherit variable with Type[*] annotation.

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#193]
**Describe the bug**
This is a bug related to #186 .

After version 1.0.54, a variable with annotation Type[X] can be inherited, but if class X's constructor signature contains keyword arguments, there might be a incorrect error report.

![image](https://user-images.githubusercontent.com/3479453/62784717-b353a980-baf1-11e9-95ab-40e596deb454.png)
_if class inherits via Type[*] variable, **kw_args in it's signature can't be handled correctly_

**Expected behavior**
The signature check can handle keyword arguments correctly

**Screenshots or Code**
```python
from typing import Type, Any, cast

some_external_lib_class = None

class A:
    """ This is the class used to "typing" an external lib's class """
    def __init__(self, **kw_args):...

alias_A: Type[A] = cast(Any, some_external_lib_class)

class B(alias_A):
    """ this class inherits class A via alias class variable"""
    pass

B(a=1) # <- this gives an error that constructor signature mismatch

class C(A):
    """ this class inherits class A directly """
    pass

C(a=1) # <- this works fine
```

**Additional context**
> **About what I wan't to achieve using the Type[*] trick**
> My goal is to use the Type[*] syntax to deal with the typing of some untyped external library, and I can't write the type stubs for the whole external library just because I need only one variable to be typed.

**Comment by erictraut** [comment]
You're trying to do a very hacky thing here. Why don't you simply create a type stub file that provides the correct interface definition for some_external_lib_class rather? That's the standard (and non-hacky) way to handle this situation.

**Comment by erictraut** [comment]
In any case, I found the cause of the error being reported, and it will be fixed in the next version of pyright.

**Comment by giyyapan** [comment]
I just found I have some misunderstanding about type stubs.

To be specific, I didn't understand how would pyright resolve them and what will happen if there's only a part of variables were typed.

A reason that I didn't consider write my own stub for a lib in the first place, is I think other dependency that also relys on it might complains about "xx is unknown import symbol" unless I define every exported variable for the lib  (which is A LOT OF WORK) .

After a while of research, I think .pyi can work just the way I need now. So thanks for your advice!

**Comment by giyyapan** [comment]
FYI, this issue can also be in "standard context".

![image](https://user-images.githubusercontent.com/3479453/62848822-b926d600-bd0f-11e9-9fb9-37398ca4dddc.png)

```python
from typing import Type, Any, cast, ClassVar

class IExternalLibClass:
    """ This is the class used to "typing" an external lib's class """
    def __init__(self, **kw_args):...

class IExternalWraperClass:
    # define a member class with type annotation only
    InnerClass: ClassVar[Type[IExternalLibClass]]

    # direct assignment, but can't add ClassVar to it's type
    InnerClass2 = IExternalLibClass

class B(IExternalWraperClass.InnerClass):
    """ this class inherits class A via alias class variable"""
    pass

B(a=1) # <- this gives an error that constructor signature mismatch

class C(IExternalWraperClass.InnerClass2):
    pass

C(a=1) # <- this works fine


class D(IExternalLibClass):
    """ this class inherits class A directly """
    pass

D(a=1) # <- this works fine

```

**Comment by erictraut** [comment]
This is now fixed in 1.0.56, which I just published.
